first_order_model/shrink_util.py

```diff
@@ -1060,43 +1109,49 @@ def compute_deletion(layer_graph,
     for mirror in layer_graph[layer].mirror:
         # Start a new custom deletion for the mirror. The only other reason is tie, which is treated a little differently.
         if reason != 'mirror':
-            custom_deletions.append((mirror, copy.copy(deletions[layer]), 'mirror'))
+            filters_to_recursively_remove.append((mirror, copy.copy(deletions[layer]), 'mirror'))
             print("Starting a mirrorred deletion")
 
     # Figure out what you need to delete from the inputs of the following convs, and the relevant portions of batchnorms.
-    # Following layers contains any layeres who can be affected by the deletion in layer_graph[layer]
+    # Following layers contains any layers who can be affected by the deletion in layer_graph[layer]
     for following_layer in following_layers:
         # If we have already addressed this layer, continue
         if following_layer in deletions:
             continue
 
         deletions[following_layer] = []
         
-        # Counts up for each index in the inputs for this layer
-        counter = 0
+        # base counts up for each index in the inputs for this layer
+        # This is to solve the following problem:
+        # If layer a, and b get concatted into the inputs of c, then we want to note that
+        # deleting from b should only affect its corresponding inputs in c.
+        # We use this index to track the "base" of an input. So if layer a has length 20, layer b has length 10
+        # Then when looking at layer b in the following loop, base=20, so you know it affects indices
+        # after 20.
+        base = 0
         for previous_layer in layer_graph[following_layer].before:
 
-            # If you don't delete from the previous layer, just increment the counter by its ouputs.
+            # If you don't delete from the previous layer, just increment the base by its ouputs.
             if previous_layer not in deletions:
-                counter += layer_graph[previous_layer].out_channels
+                base += layer_graph[previous_layer].out_channels
                 continue
 
-            # If delete from previous layer, delete corresponding element from this layer and increment the counter
+            # If delete from previous layer, delete corresponding element from this layer and increment the base
             for deletion in deletions[previous_layer]:
                 deletions[following_layer].append(
-                    (counter + deletion[0], counter + deletion[1]))
-            counter += layer_graph[previous_layer].out_channels
+                    (base + deletion[0], base + deletion[1]))
```

Comment:
I'm still really confused about the role of the `base` variable here. I think this needs much more commenting/potentially rewriting this to make more sense

Reply:
I hope my example makes it make more sense now. Leaving unresolved.